homeassistant/components/vesync/humidifier.py

```diff
@@ -0,0 +1,178 @@
+        async_dispatcher_connect(hass, VS_DISCOVERY.format(VS_HUMIDIFIERS), discover)
+    )
+
+    _setup_entities(hass.data[DOMAIN][VS_HUMIDIFIERS], async_add_entities, coordinator)
+
+
+@callback
+def _setup_entities(
+    devices: list[VeSyncBaseDevice],
+    async_add_entities,
+    coordinator: VeSyncDataCoordinator,
+):
+    """Add humidifier entities."""
+    entities = [
+        VeSyncHumidifierHA(dev, coordinator) for dev in devices if is_humidifier(dev)
+    ]
+
+    async_add_entities(entities)
+
+
+def _get_ha_mode(vs_mode: str) -> str | None:
+    ha_mode = VS_TO_HA_MODE_MAP.get(vs_mode)
+    if ha_mode is None:
+        _LOGGER.warning("Unknown mode '%s'", vs_mode)
+    return ha_mode
+
+
+def _get_vs_mode(ha_mode: str) -> str | None:
+    return HA_TO_VS_MODE_MAP.get(ha_mode)
+
+
+class VeSyncHumidifierHA(VeSyncBaseEntity, HumidifierEntity):
+    """Representation of a VeSync humidifier."""
+
+    # The base VeSyncBaseEntity has _attr_has_entity_name and this is to follow the device name
+    _attr_name = None
+
+    _attr_max_humidity = MAX_HUMIDITY
+    _attr_min_humidity = MIN_HUMIDITY
+    _attr_supported_features = HumidifierEntityFeature(HumidifierEntityFeature.MODES)
+
+    device: VeSyncHumidifierDevice
+
+    def __init__(
+        self,
+        humidifier: VeSyncHumidifierDevice,
+        coordinator: VeSyncDataCoordinator,
+    ) -> None:
+        """Initialize the VeSync humidifier device."""
+        super().__init__(humidifier, coordinator)
+
+    @property
+    def available_modes(self) -> list[str]:
+        """Return the available mist modes."""
+        return [
+            ha_mode
+            for ha_mode in (_get_ha_mode(vs_mode) for vs_mode in self.device.mist_modes)
+            if ha_mode
+        ]
+
+    @property
+    def target_humidity(self) -> int:
+        """Return the humidity we try to reach."""
+        return self.device.config["auto_target_humidity"]
+
+    @property
+    def mode(self) -> str | None:
+        """Get the current preset mode."""
+        return _get_ha_mode(self.device.details["mode"])
+
+    def set_humidity(self, humidity: int) -> None:
+        """Set the target humidity of the device."""
+        if self.device.set_humidity(humidity):
+            self.schedule_update_ha_state()
+        else:
+            raise HomeAssistantError(
+                f"An error occurred while setting humidity {humidity}."
+            )
+
+    def set_mode(self, mode: str) -> None:
+        """Set the mode of the device."""
+        if mode not in self.available_modes:
+            raise HomeAssistantError(
+                "{mode} is not one of the valid available modes: {self.available_modes}"
+            )
+        if self.device.set_humidity_mode(_get_vs_mode(mode)):
+            self.schedule_update_ha_state()
+        else:
+            raise HomeAssistantError(f"An error occurred while setting mode {mode}.")
+
+    def turn_on(
+        self,
+        **kwargs: Any,
+    ) -> None:
```

Comment:
```suggestion
    def turn_on(self, **kwargs: Any) -> None:
```

Reply:
This is due to automatic formatting in Code. I think it is coming from Ruff extension which is enabled on `Dev Container: Home Assistant Dev`. The ruff-format commit check also doesn't complain about this. Although I am not sure if it is supposed to.

Reply:
Remove the last comma and it will fix itself